### PR TITLE
Feature: Resource looker_user_attribute_user - sets a user attribute value on a looker user

### PR DESCRIPTION
**Pull request by jessicasomaiya**
## Description 

This resource adds a custom user attribute value onto a given user. When the `user_attribute` value is hidden, the Looker API returns an empty string rather than the value. This means we cannot check/verify that the `user_attribute` value is provisioned correctly in the `Read` method. 

We can check if the user attribute source is `No Value` or `Default`, meaning the value has been deleted and needs to be recreated.

This resource doesn't have an `Update` method as the `Create` method is reused in this case. If the `user_attribute_id` or the `user_id` is changed then the whole resource is destroyed and recreated. 

Last thing! The [`UserAttributeUserValues`](https://developers.looker.com/api/explorer/4.0/methods/User/user_attribute_user_values) method has a bug, so I cannot search for a specific `user_id` _and_ `user_attribute_id`. This issue has been noted and when a fix is in place we will revisit this implementation.

### Has this been tested? 

This has been tested locally against a test looker instance. Acceptance testing is almost on its way!

### CODE REVIEW

File: docs/resources/user_attribute.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Manages Looker User Attributes
     name             = "id"
     label            = "secret_id"
     data_type        = "number"
-    hidden           = true
+    hidden           = false
     default_value    = 24
     user_access      = "View"
     domain_whitelist = ["my_domain/route/sub/*"]
```
Comment (Vitzeno) [inline docs/resources/user_attribute.md:23]: Just a teeny tiny nitpick, we can't have a `domain_whitelist` whilst `hidden` is set to `false`.

File: looker/resource_looker_user_attribute_user.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	"github.com/resolutionlife/terraform-provider-looker/internal/conv"
+)
+
+func resourceUserAttributeUser() *schema.Resource {
+	return &schema.Resource{
+		Description: "This resource sets a value onto a user for the given user attribute. If a default value is already set for the user attribute, this value will override the default value. Note that if the user attribute values are hidden (can be configured when provisioning a `looker_user_attribute`) then the provider does not have the permissions to read the hidden values, and cannot verify if the value has been manually changed in the Looker UI. The provider can however check if the value has been removed, and will prompt to recreate the resource.",
+
+		CreateContext: resourceUserAttributeUserCreate,
+		ReadContext:   resourceUserAttributeUserRead,
+		UpdateContext: resourceUserAttributeUserCreate,
```
Comment (Vitzeno) [inline looker/resource_looker_user_attribute_user.go:23]: I like the clever re-use of create to also handle the value update here

Comment (jessicasomaiya) [reply looker/resource_looker_user_attribute_user.go:23]: Thank you! ☺️

**Review by Vitzeno — APPROVED** [review]
Looks great, nice work

**Pull request merged**